crates/daphne-server/src/storage_proxy_connection/kv/mod.rs

```diff
@@ -252,28 +252,74 @@ impl<'h> Kv<'h> {
     }
 
     #[tracing::instrument(
-        name = "kv_put",
+        name = "kv_delete",
         skip_all,
         fields(key, prefix = std::any::type_name::<P>()),
     )]
-    pub async fn put<P>(&self, key: &P::Key, value: P::Value) -> Result<(), Error>
+    async fn put_internal<P>(
+        &self,
+        key: &P::Key,
+        value: P::Value,
+        expiration: Option<u64>,
+    ) -> Result<(), Error>
     where
         P: KvPrefix,
         P::Key: std::fmt::Debug,
+        P::Value: Serialize,
     {
         let key = Self::to_key::<P>(key);
         tracing::debug!(key, "PUT");
-        self.http
+
+        let mut put_request = self
+            .http
             .post(self.config.url.join(&key).unwrap())
             .bearer_auth(&self.config.auth_token)
-            .body(serde_json::to_vec(&value).unwrap())
-            .send()
-            .await?
-            .error_for_status()?;
+            .body(serde_json::to_vec(&value).unwrap());
+
+        // Add expiration if provided
+        if let Some(exp) = expiration {
+            put_request = put_request.json(&serde_json::json!({ "expiration": exp }));
```

Comment:
If I'm not mistaken, this actually replaces the payload of the request. I.e., it overwrites the `.body(serde_json::to_vec(&value).unwrap())` on line 277 above. Am I reading this right?